std/algorithm/searching.d

```diff
@@ -3462,6 +3462,63 @@ unittest
 }
 
 /**
+Computes the index of the first occurrence of `range`'s minimum (respectively maximum).
+
+Params:
+    pred = The ordering predicate to use to determine the extremum (minimum or
+           maximum) element.
+    range = The input range to search.
+
+Returns:
+    The index of the first encounter of the minimum (respectively maximum) in `range`. If the
+    range is empty, -1 is returned.
+ */
+sizediff_t minIndex(alias pred = "a < b", Range)(Range range)
+    if (isForwardRange!Range && !isInfinite!Range &&
+        is(typeof(binaryFun!pred(range.front, range.front))))
+{
+    if (range.empty) return -1;
```

Comment:
I struggled with this for a bit. We have `minElement` and `minCount`, which has no reasonable way to report an empty range, so it just explodes. We then have `minPos`, which elegantly - :) - sidesteps the issue by returning a potentially empty range. This kind of leaves `minIndex` on its own. There are other functions that must deal with empty ranges, can you guys enumerate a few?